app/models/host/base.rb

```diff
@@ -175,7 +173,8 @@ def set_interfaces(parser)
       end
 
       parser.interfaces.each do |name, attributes|
-        iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
+        iface_scope = get_interface_scope(name, attributes)
```

Comment:
I think the part with `elsif`s does not have `else` which can result in nil, so if `!attributes[:virtual] && !mac_based.physical.any? && self.managed` then it can be nil. The nil should be valid return value, it indicates we didn't find matching interface and we want to create new one.